packages/web-components/src/components/ic-select/ic-select.tsx

```diff
@@ -990,6 +993,15 @@ export class Select {
     }
   };
 
+  private hasLeftIconSlot(): boolean {
+    const iconEl = this.el.querySelector(`[slot="icon"]`);
+    return iconEl !== null;
+  }
+
+  private hasValue(): boolean {
+    return this.value !== null && this.value !== "" && this.value !== undefined;
```

Comment:
```suggestion
    return !!this.value;
```

I feel like this could even be moved from a helper function, all that would be needed would be `!this.value` and `!!this.value`